Obsidian.API/_Types/HexColor.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Obsidian.API
+{
+    public struct HexColor
+    {
+
+        private string hexColor;
+
+        public HexColor(byte r, byte g, byte b)
+        {
+            hexColor = $"#{r:X2}{g:X2}{b:X2}";
+        }
+
+        public HexColor(Color color)
+        { 
+            hexColor = $"#{color.R:X2}{color.G:X2}{color.B:X2}";
+        }
+
+        public HexColor(string hex)
+        {
+            hexColor = hex;
```

Comment:
Could be a point of failure (invalid value, also missing information which format it should be), but can be addressed in a single commit afterwards.